### ISSUE TITLE
Consider: separate classes for trainable/untrainable nets

### ISSUE THREAD

**Issue by axiomizer** [axiomizer/haida-go#12]
consider creating separate classes for trainable and untrainable neural nets.

*untrainable instances will use stored batch norm stats instead of computing them for a minibatch
*trainable instances will compute batch norm stats from the minibatch and update the running averages

*trainable instances will have train() and backprop() methods
*untrainable instances will only have feedforward()

*only pickle untrainable instances

could have a "production" package that includes only untrainable nets, and a "training" package that includes training code. not sure if this is a good idea / worth doing or not
